### ISSUE TITLE
[operator] manager.targetAllocatorImage.tag non-empty default value

### ISSUE THREAD

**Issue by wand2016** [open-telemetry/opentelemetry-helm-charts#866]
Hello Team!

I want to mirror image repositories in order to avoid rate limt of DockerHub and/or ghcr.
On the other hand, I do not want to specify image tags for brevity.
Specifying both chart version and image tags is redundant.

```sh
helm install ... --set manager.collectorImage.repository=<mirror repo> # specifies only repository, NOT tag
```

This kind of installation works fine for operator and collector images because `manager.image.tag` and `manager.collectorImage.tag` has non-empty default value.

However, not working for target allocator image because `manager.targetAllocatorImage.tag` does not have non-empty default value.

https://github.com/open-telemetry/opentelemetry-helm-charts/blob/6a85b2d3f23d4f0bf79765e54c52199109a38a09/charts/opentelemetry-operator/values.yaml#L30-L38

https://github.com/open-telemetry/opentelemetry-helm-charts/blob/6a85b2d3f23d4f0bf79765e54c52199109a38a09/charts/opentelemetry-operator/templates/deployment.yaml#L49-L51

Therefore I want `manager.targetAllocatorImage.tag` to have non-empty default value like `manager.image.tag` or `manager.collectorImage.tag`.

Thanks!

**Comment by TylerHelmuth** [comment]
Yes we can provide a default value for these, although instead of specifying the image version in the values.yaml we should really be using the appVersion.
